File: in_Java/0690 Employee Importance.java
/*
// Definition for Employee.
class Employee {
    public int id;
    public int importance;
    public List<Integer> subordinates;
};
*/

class Solution {
    public int getImportance(List<Employee> employees, int id) {
        HashMap<Integer,Integer> hm1 = new HashMap<>();
        HashMap<Integer,List<Integer>> hm2 = new HashMap<>();
        for(Employee e : employees){
            int k = e.id;
            int v = e.importance;
            List<Integer> L = e.subordinates;
            hm1.put(k,v);
            hm2.put(k,L);
        }
        int res = 0;
        Set<Integer> visited = new HashSet<>();
        visited.add(id);
        Queue<Integer> q = new LinkedList<>();
        q.add(id);
        int n;
        int cur;
        while(q.size() != 0){
            n = q.size();
            while(n-- != 0){
                cur = q.poll();
                res += hm1.get(cur);
                for(int i : hm2.get(cur)){
                    if(!visited.contains(i)){
                        q.add(i);
                        visited.add(i);
                    }
                }
            }
        }
        return res;
    }
}

File: in_Python/0690 Employee Importance.py
"""
# Definition for Employee.
class Employee(object):
    def __init__(self, id, importance, subordinates):
        #################
        :type id: int
        :type importance: int
        :type subordinates: List[int]
        #################
        self.id = id
        self.importance = importance
        self.subordinates = subordinates
"""

class Solution(object):
    def getImportance(self, employees, id):
        """
        :type employees: List[Employee]
        :type id: int
        :rtype: int
        """
        for e in employees:
            if e.id == id:
                row = e
                break
        res = row.importance
        st = row.subordinates
        visited = []
        while len(st) != 0:
            nextem = st.pop(0)
            if nextem not in visited:
                visited.append(nextem)
                for e in employees:
                    if e.id == nextem:
                        t = e
                        break
                res += t.importance
                for x in t.subordinates:
                    st.append(x)
        return res
            
        
        